components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -431,23 +432,53 @@ - (CGFloat)justifiedWidth {
 - (void)scrollUntilSelectedItemIsVisibleWithoutAnimation {
   NSUInteger index = [self.items indexOfObject:self.selectedItem];
   if (index == NSNotFound || index >= self.containerView.arrangedSubviews.count) {
+    index = 0;
+  }
+  if (self.containerView.arrangedSubviews.count == 0U) {
     return;
   }
 
-  CGFloat selectedItemOriginX = 0;
+  CGRect estimatedItemFrame = [self estimatedFrameInContainerViewForItemAtIndex:index];
+  [self scrollRectToVisible:[self convertRect:estimatedItemFrame fromView:self.containerView]
+                   animated:NO];
+}
+
+- (CGRect)estimatedFrameInContainerViewForItemAtIndex:(NSUInteger)index {
+  if (index == NSNotFound || index >= self.containerView.arrangedSubviews.count) {
+    return CGRectZero;
+  }
+
+  BOOL isRTL =
+      [self mdf_effectiveUserInterfaceLayoutDirection] == UIUserInterfaceLayoutDirectionRightToLeft;
+  CGFloat viewOriginX =
+      isRTL ? CGRectGetMaxX(self.containerView.bounds) : CGRectGetMinX(self.containerView.bounds);
+
   for (NSUInteger i = 0; i < index; ++i) {
-    CGSize expectedItemSize = [self expectedSizeForView:self.containerView.arrangedSubviews[i]];
-    selectedItemOriginX += expectedItemSize.width;
+    CGSize viewSize = [self expectedSizeForView:self.containerView.arrangedSubviews[i]];
+    if (isRTL) {
+      viewOriginX -= viewSize.width;
+    } else {
+      viewOriginX += viewSize.width;
+    }
   }
-  CGSize expectedSelectedItemSize =
-      [self expectedSizeForView:self.containerView.arrangedSubviews[index]];
-  CGRect expectedItemFrame =
-      CGRectMake(CGRectGetMinX(self.bounds) + selectedItemOriginX, CGRectGetMinY(self.bounds),
-                 expectedSelectedItemSize.width, expectedSelectedItemSize.height);
-  [self scrollRectToVisible:expectedItemFrame animated:NO];
+  CGSize viewSize = [self expectedSizeForView:self.containerView.arrangedSubviews[index]];
+  if (isRTL) {
+    viewOriginX -= viewSize.width;
+  }
+  CGRect itemFrameInContainerView = CGRectMake(
+      viewOriginX, CGRectGetMinY(self.containerView.bounds), viewSize.width, viewSize.height);
+  CGRect convertedRect = [self convertRect:itemFrameInContainerView fromView:self.containerView];
+  return convertedRect;
 }
 
 - (CGSize)expectedSizeForView:(UIView *)view {
+  if (self.containerView.distribution == UIStackViewDistributionFillEqually) {
+    if (CGRectGetWidth(self.containerView.bounds) < 0) {
```

Comment:
What is the case that the with can be less than 0?

Reply:
Ah, that should be `<=`. I'll fix it, thanks!

Reply:
Add a TODO because I couldn't reach this code (with either condition) during simulator testing. However, I'm going to leave it in for now.